R/arg.R

```diff
@@ -47,30 +46,42 @@ arg_match <- function(arg, values = NULL) {
   }
 
   arg <- arg[[1]]
-  i <- match(arg, values)
+  arg_match0(arg, values, arg_nm)
+}
 
-  if (is_na(i)) {
-    msg <- arg_match_invalid_msg(arg_nm, values)
+#' @description
+#' `arg_match0()` is a bare-bones version if performance is at a premium.
+#' It requires a string as `arg` and explicit `values`.
+#' @param arg_nm The label to be used for `arg` in error messages.
+#' @rdname arg_match
+#' @export
+#' @examples
+#'
+#' # Use the bare-bones version with explicit values for speed:
+#' arg_match0("bar", c("foo", "bar", "baz"))
+arg_match0 <- function(arg, values, arg_nm = as_label(substitute(arg))) {
+  .External2(rlang_ext2_arg_match0, arg, values, arg_nm)
```

Comment:
Should use `.External()` and pass the environment, even if slightly slower. `arg_nm` doesn't need to be passed down since it's constant.

Reply:
Can you propose a change please?

I can benchmark before-after to assess impact, unless you have already?

Reply:
I did once, the impact was minimal. Even if not minimal, I still would like to minimise the use of .External2 because it's not a public entry point and because it's tricky to use. Any call into R that changes the value of `R_Visibility` may cause unexpected visibility return.

You can leave this as is and I'll push a change to .External once the PR is done.